src/main.rs

```diff
@@ -109,3 +117,110 @@ pub fn get_zones(pbf: &mut OsmPbfReader) {
         }
     }
 }
+
+pub fn read_libpostal_yaml_folder(yaml_files_folder: String) -> BTreeMap<String, model::Country> {
+    let paths = fs::read_dir(yaml_files_folder).expect("The yaml directory doesn't exist.");
+    let mut countries: BTreeMap<String, model::Country> = BTreeMap::new();
+
+    for entry in paths {
+        let mut contents = String::new();
+        let a_path = entry.unwrap().path();
+
+        File::open(&a_path)
+            .unwrap()
+            .read_to_string(&mut contents)
+            .expect("Something went wrong reading the file");
+
+        let deserialized_country = match read_libpostal_yaml(&contents) {
+            Ok(a) => a,
+            Err(_) => continue,
+        };
+
+        let reference = a_path
+            .file_name()
+            .unwrap()
+            .to_str()
+            .unwrap()
+            .get(0..2)
```

Comment:
I think we can do better :laughing: like https://doc.rust-lang.org/std/path/struct.Path.html#method.file_name